cli/mender-artifact/copy.go

```diff
@@ -57,34 +57,44 @@ func Copy(c *cli.Context) (err error) {
 
 	var r io.ReadCloser
 	var w io.WriteCloser
+	wclose := func(w io.Closer) {
+		cerr := w.Close()
+		if err == nil {
+			err = cerr
+		}
+	}
+	var vfile VPFile
 	switch parseCLIOptions(c) {
 	case copyin:
 		r, err = os.OpenFile(c.Args().First(), os.O_CREATE|os.O_RDONLY, 0655)
 		defer r.Close()
 		if err != nil {
 			return cli.NewExitError(fmt.Sprintf("%v", err), 1)
 		}
-		w, err = NewPartitionFile(comp, c.Args().Get(1), c.String("key"))
+		vfile, err = virtualPartitionFile.Open(comp, c.Args().Get(1), c.String("key"))
+		defer wclose(vfile)
 		if err != nil {
-			w.Close()
 			return cli.NewExitError(fmt.Sprintf("%v", err), 1)
 		}
+		w = vfile
 	case copyinstdin:
 		r = os.Stdin
-		w, err = NewPartitionFile(comp, c.Args().First(), c.String("key"))
+		vfile, err = virtualPartitionFile.Open(comp, c.Args().First(), c.String("key"))
+		defer wclose(vfile)
 		if err != nil {
-			w.Close()
 			return cli.NewExitError(fmt.Sprintf("%v", err), 1)
 		}
+		w = vfile
 	case copyout:
-		r, err = NewPartitionFile(comp, c.Args().First(), c.String("key"))
-		defer r.Close()
+		vfile, err = virtualPartitionFile.Open(comp, c.Args().First(), c.String("key"))
+		defer vfile.Close()
```

Comment:
`defer wclose`

Reply:
Other places out of Copy and Delete are missing this improvement, but it is not a big deal so I won't insist more. Only this one I insist for the sake of consistency inside the same function :smiley: